docs/content/mailru.md

```diff
@@ -0,0 +1,274 @@
+because files are searched by hash in all accounts of all mailru users.
+Please note that rclone needs local memory and disk space to calculate
+content hash in advance and decide whether full upload is required.
+Also, if rclone does not know file size in advance (e.g. in case of
+streaming or partial uploads), it will not even try this optimization.
+
+- Config:      speedup_enable
+- Env Var:     RCLONE_MAILRU_SPEEDUP_ENABLE
+- Type:        bool
+- Default:     true
+- Examples:
+    - "true"
+        - Enable
+    - "false"
+        - Disable
+
+### Advanced Options
+
+Here are the advanced options specific to mailru (Mail.ru Cloud).
+
+#### --mailru-speedup-file-patterns
+
+Comma separated list of file name patterns eligible for speedup (put by hash).
+Patterns are case insensitive and can contain `*` or `?` meta characters.
+
+- Config:      speedup_file_patterns
+- Env Var:     RCLONE_MAILRU_SPEEDUP_FILE_PATTERNS
+- Type:        string
+- Default:     "*.mkv,*.avi,*.mp4,*.mp3,*.zip,*.gz,*.rar,*.pdf"
+- Examples:
+    - `""`
+        - Empty list completely disables speedup (put by hash).
+    - `"*"`
+        - All files will be attempted for speedup.
+    - `"*.mkv,*.avi,*.mp4,*.mp3"`
+        - Only common audio/video files will be tried for put by hash.
+    - `"*.zip,*.gz,*.rar,*.pdf"`
+        - Only common archives or PDF books will be tried for speedup.
+
+#### --mailru-speedup-max-disk
+
+This option allows you to disable speedup (put by hash) for large files
+(because preliminary hashing can exhaust you RAM or disk space)
+
+- Config:      speedup_max_disk
+- Env Var:     RCLONE_MAILRU_SPEEDUP_MAX_DISK
+- Type:        SizeSuffix
+- Default:     3G
+- Examples:
+    - "0"
+        - Completely disable speedup (put by hash).
+    - "1G"
+        - Files larger than 1Gb will be uploaded directly.
+    - "3G"
+        - Choose this option if you have less than 3Gb free on local disk.
+
+#### --mailru-speedup-max-memory
+
+Files larger than the size given below will always be hashed on disk.
+
+- Config:      speedup_max_memory
+- Env Var:     RCLONE_MAILRU_SPEEDUP_MAX_MEMORY
+- Type:        SizeSuffix
+- Default:     32M
+- Examples:
+    - "0"
+        - Preliminary hashing will always be done in a temporary disk location.
+    - "32M"
+        - Do not dedicate more than 32Mb RAM for preliminary hashing.
+    - "256M"
+        - You have at most 256Mb RAM free for hash calculations.
+
+#### --mailru-check-hash
+
+What should copy do if file checksum is mismatched or invalid
+
+- Config:      check_hash
+- Env Var:     RCLONE_MAILRU_CHECK_HASH
+- Type:        bool
+- Default:     true
+- Examples:
+    - "true"
+        - Fail with error.
+    - "false"
+        - Ignore and continue.
+
+#### --mailru-user-agent
+
+User agent used by client (internal)
+
+- Config:      user_agent
+- Env Var:     RCLONE_MAILRU_USER_AGENT
+- Type:        string
+- Default:     "CloudDiskOWindows 17.12.0009 beta WzBbt1Ygbm"
```

Comment:
Is this necessary?  I prefer to use rclone's user agent if possible so cloud providers can identify problems, however if this is the only way then fine!

Reply:
I will recheck whether API works stable for non-mailru user agent.
Is there a generic flag to specify non-rclone user agent?

Reply:
This is the generic one

      --user-agent string   Set the user-agent to a specified string. The default is rclone/ version (default "rclone/v1.49.1-DEV")

Reply:
Can a generic flag like this be attached in the [remote] section?

Reply:
I can't find a way to stick `--user-agent` to remote in rclone.conf. I will then keep `--mailru-user-agent` but make it by default empty and fall back to --user-agent.